/run/secres is a emerging container best practice for depolying secrets
to applications. With this patch, Odoo will happily read theadmin password
from the /run/secrets/adminpwd location.

 odoo/tools/config.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/odoo/tools/config.py b/odoo/tools/config.py
index d3906900c3d..468f0e9c1e5 100644
--- a/odoo/tools/config.py
+++ b/odoo/tools/config.py
@@ -64,10 +64,15 @@ class configmanager(object):
                       from Python code without resorting to environment
                       variable
         """
+        # Check for a bind-mounted secrets file
+        try:
+            admin_pwd_file = open("/run/secrets/adminpwd","r").read().splitlines()[0]
+        except:
+            admin_pwd_file = None
         # Options not exposed on the command line. Command line options will be added
         # from optparse's parser.
         self.options = {
-            'admin_passwd': 'admin',
+            'admin_passwd': admin_pwd_file and admin_pwd_file or 'admin',
             'csv_internal_sep': ',',
             'publisher_warranty_url': 'http://services.openerp.com/publisher-warranty/',
             'reportgz': False,
 
